### DISCUSSION TITLE
Particle insert

### DISCUSSION THREAD

**Discussion by Er9y714** [firemodels/fds#11639] [Q&A] [answered]
Hi

I insert particles into 1m^3 domain with   MASS_PER_VOLUME=1.0E-5. I expect to get 1E-5 kg total mass of this species. However I get  9.49E-06 kg.  (5% error). 

Is that error expected? How can I improve the result?

Thank you. 

```

&HEAD CHID='insert'/
&TIME T_END=30.0/
&DUMP DT_DEVC=1.0E-3, DT_SLCF=0.01, SMOKE3D=.FALSE., DT_PART=0.01/
&MISC Y_CO2_INFTY=0.0/
&RADI RADIATION=.FALSE./

&MESH ID='Mesh01-01', IJK=10,10,10, XB=0.0,1.0,0.0,1.0,0.0,1.0/

&SPEC ID='HYDROGEN CYANIDE'/

&PART ID='Particle01',
      SPEC_ID='HYDROGEN CYANIDE',
      DIAMETER=100.0,
      MONODISPERSE=.TRUE.,
      AGE=1800.0,
      SAMPLING_FACTOR=1/


&INIT ID='Particle Cloud', PART_ID='Particle01', N_PARTICLES=1000, MASS_PER_VOLUME=1.0E-5, XB=0.0,1.0,0.0,1.0,0.0,1.0/ 

&DEVC ID='HCN Density_VOLUME INTEGRAL', QUANTITY='DENSITY', SPEC_ID='HYDROGEN CYANIDE', SPATIAL_STATISTIC='VOLUME INTEGRAL', XB=0.0,1.0,0.0,1.0,0.0,1.0/

&TAIL /

```

**Comment by drjfloyd** [comment, marked as answer]
You are inserting liquid particles that are evaporating. The DEVC is measuring the gas phase. The particles take a few timesteps to evaporate and as they evaporate they fall. The default floor condition is a porous floor for particles. Some are falling through before they evaporate. Add POROUS_FLOOR=T to MISC and the error will drop. 

There will be some numerical error which also seems to be an impact. With your grid size the time step starts off at 0.16 s and the particles are evaporating in a few time steps. If you add DT to TIME and set smaller and smaller DT you will see the error drops as DT is smaller.

If you add these lines you can see that the correct mass is insereted:

&PROP ID='PDPA2', PDPA_RADIUS=2, QUANTITY='MASS CONCENTRATION', PART_ID='Particle01',
      PDPA_INTEGRATE=.FALSE., PDPA_NORMALIZE=.FALSE. /
&DEVC XYZ=0.5,0.5,0.5, QUANTITY='PDPA', PROP_ID='PDPA2', ID='m_HCN', UNITS='kg'/

**Comment by rmcdermo** [comment]
Another option is to just add `STATIC=T` to the PART line and the result is spot on.

**Comment by rmcdermo** [comment]
@drjfloyd One thing that surprised me about this case is that the only MATL that is output to the .out file is AIR.  So, my first thought was that maybe AIR was being used as the PART solid material and we were grabbing that density accidentally.  Should we not have a MATL line for the 'HYDROGEN CYANIDE' in this case?

**Comment by drjfloyd** [comment]
HYDROGEN CYANIDE is a predefined species with predefined liquid properties. When you assign only a SPEC_ID to a PART, FDS treats it as a liquid that evaporates. If you don't have predefined liquid properties, FDS will throw an error. 

Not sure why AIR is being output as a MATL. Some logic error somewhere.

**Comment by drjfloyd** [comment]
It is this code in read.  AIR is added automatically to support simpler vegatation definitions. Not a logic error, just someting I didn't realize was happening.

! Add reserved materials if necessary

N_MATL_RESERVED = 2
ALLOCATE(SEARCH_PHRASE(N_MATL_RESERVED)) ; ALLOCATE(MATL_NAME_RESERVED(N_MATL_RESERVED))
SEARCH_PHRASE(1) = 'MOISTURE_FRACTION' ; MATL_NAME_RESERVED(1) = 'MOISTURE'
SEARCH_PHRASE(2) = 'MASS_PER_VOLUME'   ; MATL_NAME_RESERVED(2) = 'AIR'

**Comment by rmcdermo** [comment]
Thanks.  My thinking is that anything using the thermally thick model should explicitly write out whatever MATL it is using, no?

**Comment by drjfloyd** [comment]
This isn't using the thermally thick model. This is like a water droplet from a sprinkler, only it is HCN and not H2O.

**Reply by rmcdermo** [reply]
OK, thanks

**Comment by drjfloyd** [comment]
We are writing the liquid data:

```
   Liq. Enthalpy (J/kg)     Melt  260 K:  3.907E+06
                                  279 K:  3.954E+06
                            Boil  299 K:  3.990E+06
 
   Liq. Spec. Heat (J/kg/K) Melt  260 K:  2.182E+03
                                  279 K:  2.182E+03
                            Boil  299 K:  2.182E+03
 
   Heat of Vapor. (J/kg)    Melt  260 K:  9.680E+05
                                  279 K:  9.509E+05
                            Boil  299 K:  9.336E+05
```

**Comment by Er9y714** [comment]
Thank you very much. I totally forgot about droplets disappearing on the floor by default.
